### ISSUE TITLE
Add documentation for classifiers, including model training data

### ISSUE THREAD

**Issue by sjspielman** [AlexsLemonade/mbSSP#10]
In #3, we'll be adding the classifiers as package datasets. This issue tracks fleshing out the documentation for these objects with information about model training procedures and data sources.

**Comment by sjspielman** [comment]
Want to check in with @envest to confirm some details here based on what I've seen in the other repo. Can you let me know if this is correct or not? Thanks!

- Models were built with this script:  https://github.com/AlexsLemonade/medulloblastoma-classifier/blob/main/predict/predict_baseline_models.R
- The model _only uses_ samples in `GSE37418` as training data. This is how the script seems to currently be coded - does this indeed correspond to the model objects currently in use?
  - Source: https://github.com/AlexsLemonade/medulloblastoma-classifier/blob/941b66afc54019b62ceeb3018c3a02f2841b14ee/predict/predict_baseline_models.R#L97-L116 & https://github.com/AlexsLemonade/medulloblastoma-classifier/blob/941b66afc54019b62ceeb3018c3a02f2841b14ee/predict/predict_baseline_models.R#L126-L145
  - In addition, can you confirm if the parameters in the lines linked above correspond to the existing model objects? 
  - This training input data was obtained from https://www.refine.bio/experiments/GSE37418/novel-mutations-target-distinct-subgroups-of-medulloblastoma and the matrix was directly inputted to the modeling code, e.g. no data transformations were applied 

A general thought I also had, it would be good to tag the `AlexsLemonade/medulloblastoma-classifier` to where the code definitely corresponds to the models. This will be helpful for adding permalinks as part of this package's documentation.

**Comment by jaclyn-taroni** [comment]
> * The model _only uses_ samples in `GSE37418` as training data.

This is not correct. If you look at the underlying functions, that argument says make sure you _always include this array data_ in the training set, not only use this array data as training data.

https://github.com/AlexsLemonade/medulloblastoma-classifier/blob/941b66afc54019b62ceeb3018c3a02f2841b14ee/utils/modeling.R#L39

https://github.com/AlexsLemonade/medulloblastoma-classifier/blob/941b66afc54019b62ceeb3018c3a02f2841b14ee/utils/modeling.R#L215-L232

This allows for comparison to `medulloPackage` in all repeats.

You can see what data was used in training by grabbing the `training_metadata` elements of the baseline models list. So, if you find `training_metadata` for the "official repeat," you can grab the study identifiers from there.

**Comment by sjspielman** [comment]
Thanks, @jaclyn-taroni! Will dig deeper.

**Comment by sjspielman** [comment]
_Actually looking at the model_, there are 407 samples in the training data. Some of them can be easily communicated with just the Study ID, but not the St. Jude data. It would make the docs hard to read if we included all the ids in it, so I think it would make sense to add a package data frame with the training metadata. This TSV stored in the model looks like this:

```
   sample_accession subgroup     study is_duplicate platform
1          GSM260959      WNT  GSE10327        FALSE    Array
2          GSM260960       G4  GSE10327        FALSE    Array
3          GSM260961      SHH  GSE10327        FALSE    Array
```

We should remove the `is_duplicate` column and rename `study` to something more general like `source`, and add this to `data/` and document it. Then, we can refer to this package variable in the classifier docs. Overall, this increases training data transparency I think.

**Comment by envest** [comment]
> * Models were built with this script:  https://github.com/AlexsLemonade/medulloblastoma-classifier/blob/main/predict/predict_baseline_models.R

Yes

> * The model _only uses_ samples in `GSE37418` as training data. This is how the script seems to currently be coded - does this indeed correspond to the model objects currently in use?

`GSE37418` is forced to be included as training data but others may also be assigned to training (see Jackie's comment). We wanted to always include GSE37418 in training because medulloPackage was also trained on it, so testing medulloPackage on it was not best practice ([see original issue here](https://github.com/AlexsLemonade/medulloblastoma-classifier/issues/63)).

>   * In addition, can you confirm if the parameters in the lines linked above correspond to the existing model objects?

Yes

>   * This training input data was obtained from https://www.refine.bio/experiments/GSE37418/novel-mutations-target-distinct-subgroups-of-medulloblastoma and the matrix was directly inputted to the modeling code, e.g. no data transformations were applied

Here's the specific place way all refine.bio data are processed:
- downloaded from here: https://github.com/AlexsLemonade/medulloblastoma-classifier/blob/main/00-download_data.sh#L63-L69
- metadata came from this harmonization study: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE124814
- each dataset was read in here with no normalization: https://github.com/AlexsLemonade/medulloblastoma-classifier/blob/main/02-gather_data.R#L119-L129


> A general thought I also had, it would be good to tag the `AlexsLemonade/medulloblastoma-classifier` to where the code definitely corresponds to the models. This will be helpful for adding permalinks as part of this package's documentation.

Do you mean like versioning the code? i.e. "cut a release" :tm:

> We should remove the is_duplicate column and rename study to something more general like source, and add this to data/ and document it. Then, we can refer to this package variable in the classifier docs. Overall, this increases training data transparency I think.

Happy with increasing transparency, but there may be a problem/create confusion if we change the column names. The `check_input_data()` function in `utils/modeling.R` expects certain metadata column names, including `sample_accession`, `study`, `subgroup`, and `platform`. https://github.com/AlexsLemonade/medulloblastoma-classifier/blob/main/utils/modeling.R#L509-L536

Thank you, @sjspielman!
